### PR TITLE
Material centric2

### PR DESCRIPTION
**Pull request by nathanielmorgan**
# Description
The new material centric implementation is up and running on CPUs and GPUs.  This version loops over the elements in the mesh that have a material (key to eliminating thread divergence).  The material centric data structures also allow developers to append data to the end of the material storage, key for any future remap research (for ALE) or future Eulerian AMR research.  The coding also solves an energy equation per material.

This version also renames variables (more of them) according to the google style guide and reorganizes the files for cleanliness.  

## Type of change

Please select all relevant options

- [x] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Formatting and/or style fixes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Ran tests on CPUs and GPUs

- [x] Sedov : 
- [x] Sedov-origin-erosion : 
- [x] impact-erosion : 

**Test Configuration**:
* OS version: Mac OS 
* Hardware: x86 CPU and NVidia GPU
* Compiler: clang with openMP backend

* OS version: Linux, 
* Hardware: x86 CPU with a V100 GPU
* Compiler: gcc plus CUDA

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [0] My changes generate a few cuda warnings about unused variables, those variables for needed
- [x] The code builds from scratch with my new changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jacob-moore22** [comment]
Just a note to any reviewers.  We changed the cmake to do a more monolithic build because we were having issues with compilation.  In the end, our issues were probably not from creating libraries and then linking them. We may revert back at some point, but only after further consideration and changes to how we handle everything in the "Common" directory.

File: single-node-refactor/src/Solvers/SGH_solver/src/time_integration.cpp [hunk 1]
```diff
@@ -168,11 +177,15 @@ void SGH::get_timestep(mesh_t& mesh,
         }
 
         // local dt calc based on CFL
-        double dt_lcl_ = dt_cfl * dist_min / (MaterialPoints_sspd(elem_gid) + fuzz);
+        double dt_lcl_ = dt_cfl * dist_min / (MaterialPoints_sspd(mat_elem_lid) + fuzz);
+
+        if (MaterialToMeshMaps_elem(mat_elem_lid) == true){
```
Comment (jacob-moore22) [inline single-node-refactor/src/Solvers/SGH_solver/src/time_integration.cpp:182]: I dont understand this if check.  What is the comparison here?

File: single-node-refactor/src/common/io_utils.h [hunk 2]
```diff
@@ -97,7 +158,6 @@ class MeshReader
     ///
     /////////////////////////////////////////////////////////////////////////////
     void read_mesh(mesh_t& mesh, 
-                   MaterialPoint_t& MaterialPoints,
```
Comment (jacob-moore22) [inline single-node-refactor/src/common/io_utils.h:100]: What drove the removal of material points here?

Comment (nathanielmorgan) [reply single-node-refactor/src/common/io_utils.h:100]: the mesh doesn't have material info

**Review by jacob-moore22 — APPROVED** [review]
lgtm

**Comment by nathanielmorgan** [comment]
Material centric implementation

**Pull request merged**